File: menu/javascript/SimpleTag.js

function SimpleTag(posX, posY){
	this.position = [posX, posY];
	this.size = [50, 50];
	this.sharkId = 0;
}

SimpleTag.prototype.getSharkId = function(){
	return this.sharkId = 0;
}

SimpleTag.prototype.setSharkId = function(id){
	this.sharkId = id;
}

SimpleTag.prototype.getSize = function(){
	return this.size;
}

SimpleTag.prototype.setSize = function(X, Y){
	this.size = [X, Y];
}

/*
 * Return true or false depending if the thresold is bigger than the distance 
 * between the two vectors
 */
SimpleTag.prototype.overlap = function(other, threshold){
	var a1 = (this.position[0]-other.position[0]);
	var b1 = (this.position[1]-other.position[1]);
	var dst1 = Math.sqrt(a1*a1 + b1*b1);
	//var dst1 = dst(this.position, other.position)

	var a2 = (this.position[0]+this.size[0])-(other.position[0]+other.size[0]);
	var b2 = (this.position[1]+this.size[1])-(other.position[1]+other.size[1]);
	var dst2 = Math.sqrt(a2*a2 + b2*b2);
	//var dst2 = dst(this.position+this.size, other.position+other.size )

	// Compare the distance between the the two vectors and vectors + size with the threshold value
	if((dst1 <= threshold) && (dst2 <= threshold)){
		return true;
	}
	else{
		return false;
	}
}

/*
 * Give the distance between two points
 */
function dst(point1, point2){
	var a1 = (point1[0]-point2[0]);
	var b1 = (point1[1]-point2[1]);
	return = Math.sqrt(a1*a1 + b1*b1);
}

module.exports = SimpleTag;

File: Shark-Tagging-Game/core/src/com/dcu/sharktag/SimpleTag.java
package com.dcu.sharktag;

import com.badlogic.gdx.math.Vector2;

/*
 * This class gets serialised and sent to the server as a tag
 */
public class SimpleTag {

	protected Vector2 position = new Vector2();
	protected Vector2 size = new Vector2();
	protected int sharkId = 0;
	
	public SimpleTag(float x, float y){
		position = new Vector2(x, y);
		size = new Vector2(50, 50);
	}
	
	public int getSharkId(){
		return sharkId;
	}
	
	public void setSharkId(int id){
		sharkId = id;
	}
	
	public Vector2 getSize(){
		return size;
	}
	
	public void setSize(Vector2 s){
		size = s;
	}
	
	public boolean overlap(SimpleTag other, float threshold){
		Vector2 tmpPos = new Vector2(position);
		Vector2 tmpPos2 = new Vector2(other.position);
		
		if(tmpPos2.dst(tmpPos) <= threshold &&
				(tmpPos2.add(other.size)).dst(tmpPos.add(size)) <= threshold){
			return true;
		}
		else{
			return false;
		}
	}
}
